packages/shared/src/components/modals/OnboardingModal.tsx

```diff
@@ -0,0 +1,150 @@
+  const [invalidMessage, setInvalidMessage] = useState<string>(null);
+  const { onboardingSteps, onboardingFiltersLayout, onboardingMinimumTopics } =
+    useContext(FeaturesContext);
+  const { insaneMode, themeMode, setTheme, toggleInsaneMode } =
+    useContext(SettingsContext);
+  const { tagsCategories } = useFeedSettings();
+  const { onFollowTags, onUnfollowTags } = useTagAndSource({
+    origin: 'filter onboarding',
+  });
+  const [step, setStep] = useState(0);
+  const onStepChange = (beforeStep: number, stepNow: number) => {
+    setStep(stepNow);
+    const filterStep = onboardingSteps.indexOf(OnboardingStep.Topics);
+
+    if (beforeStep - 1 === filterStep) {
+      const key = getFeedSettingsQueryKey();
+      const { feedSettings } = client.getQueryData<AllTagCategoriesData>(key);
+      updateLocalFeedSettings(feedSettings);
+    }
+  };
+
+  const onChangeSelectedTopic = (value: string) => {
+    const isFollowed = !selectedTopics[value];
+    const tagCommand = isFollowed ? onFollowTags : onUnfollowTags;
+    const { tags, title } = tagsCategories.find(({ id }) => id === value);
+    tagCommand({ tags, category: title });
+    setSelectedTopics({ ...selectedTopics, [value]: isFollowed });
+    if (invalidMessage) {
+      setInvalidMessage(null);
+    }
+  };
+
+  const components: Record<OnboardingStep, ReactNode> = {
+    topics: (
+      <FilterOnboarding
+        key={OnboardingStep.Topics}
+        topicLayout={onboardingFiltersLayout}
+        selectedId={selectedTopics}
+        tagsCategories={tagsCategories}
+        onSelectedChange={onChangeSelectedTopic}
+      />
+    ),
+    layout: (
+      <LayoutOnboarding
+        key={OnboardingStep.Layout}
+        isListMode={insaneMode}
+        onListModeChange={toggleInsaneMode}
+      />
+    ),
+    theme: (
+      <ThemeOnboarding
+        key={OnboardingStep.Theme}
+        selectedTheme={themeMode}
+        onThemeChange={setTheme}
+      />
+    ),
+  };
+
+  const onValidateFilter = () => {
+    const selected = Object.values(selectedTopics).filter((value) => !!value);
+    const isValid = selected.length >= onboardingMinimumTopics;
+    const errorMessage = `Choose at least ${onboardingMinimumTopics} topics to follow`;
+    setInvalidMessage(isValid ? null : errorMessage);
+
+    return isValid;
+  };
+
+  const nextButtonValidations = (() => {
+    const validations = Array(STEPS_COUNT).fill(null);
+    const filterStep = onboardingSteps.indexOf(OnboardingStep.Topics);
+
+    if (filterStep !== -1) {
+      validations[filterStep + 1] = onValidateFilter;
+    }
+
+    return validations;
+  })();
+
+  const onFinishOnboarding = async (isLogin: boolean) => {
+    const key = getFeedSettingsQueryKey();
+    const { feedSettings } = client.getQueryData<AllTagCategoriesData>(key);
+    const { hasFilters } = await registerLocalFilters(feedSettings);
+    if (!hasFilters) {
+      return;
+    }
+
+    await refetchBoot();
```

Comment:
Why is it required?

Reply:
After logging in, the local state is not yet updated that the user has successfully logged in (with email and password). Either we will have to do it at the `AuthOptions` component or here.

Reply:
Seems like there is just an issue with the local environment and `AuthOptions` already fetches the boot data when necessary.